### PR TITLE
Javascript javascript1 week3

### PR DESCRIPTION
**Pull request by RodrigoGuinsbergPinto**
Hi, here is my JS-week3 homework.
There are 2 exercises in the when will we be there file (when will we be there exercise and note app exercise).
Please ignore file '?'. I don't know how to delete it.

### CODE REVIEW

File: javascript/javascript1/week2/js-week2.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
+// FLIGHT BOOKING FULLNAME FUNCTION
```
Comment (cecastosic) [inline javascript/javascript1/week2/js-week2.js]: This is homework for week2 which should be in a different PR.

File: javascript/javascript1/week3/js1-week3.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Homework - JS1-week3</title>
+  </head>
+  <body>
+    <script src="item array removal.js"></script>
+    <script src="when will we be there.js"></script>
+    <script src="series duration.js"></script>
+    <script src="note app.js"></script>
```
Comment (cecastosic) [inline javascript/javascript1/week3/js1-week3.html]: Check file names `itemArrayRemoval.js` etc

File: javascript/javascript1/week3/itemArrayRemoval.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "katrine",
+  "Tala",
+];
+
+const nameToRemove = "Ahmad";
+
+const indexOfNameToRemove = names.indexOf(nameToRemove);
+names.splice(indexOfNameToRemove, 1);
+
+console.log(names);
```
Comment (cecastosic) [inline javascript/javascript1/week3/itemArrayRemoval.js:24]: Cool! Can you also write a function that would do the same? Thanks!

File: javascript/javascript1/week3/whenWillWeBeThere.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// WHEN WILL WE BE THERE
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+  
+const travelTimeInMinutes = travelInformation.destinationDistance * 60 / travelInformation.speed;
+  
+function convert(num) {
```
Comment (cecastosic) [inline javascript/javascript1/week3/whenWillWeBeThere.js]: I would suggest to make a function 
```suggestion
function getTotalTripTime(information) {
```
which will take information as a parameter and return the calculations (as a sentence) 
In that case `const travelTimeInMinutes = travelInformation.destinationDistance * 60 / travelInformation.speed;` should be inside the function.

File: javascript/javascript1/week3/whenWillWeBeThere.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    const hours = Math.floor(num / 60);
+    const minutes = Math.floor(num % 60);
+    return (`The total time of the trip will be ${hours} hours and ${minutes} minutes.`);
+}
+  
+convert(travelTimeInMinutes);
+console.log(convert(travelTimeInMinutes));
```
Comment (cecastosic) [inline javascript/javascript1/week3/whenWillWeBeThere.js]:
```suggestion
console.log(getTotalTripTime(travelInformation));
```

File: javascript/javascript1/week3/whenWillWeBeThere.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  
+const travelTimeInMinutes = travelInformation.destinationDistance * 60 / travelInformation.speed;
+  
+function convert(num) {
+    const hours = Math.floor(num / 60);
+    const minutes = Math.floor(num % 60);
+    return (`The total time of the trip will be ${hours} hours and ${minutes} minutes.`);
+}
+  
+convert(travelTimeInMinutes);
```
Comment (cecastosic) [inline javascript/javascript1/week3/whenWillWeBeThere.js]: not needed

File: javascript/javascript1/week3/whenWillWeBeThere.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+  
+const travelTimeInMinutes = travelInformation.destinationDistance * 60 / travelInformation.speed;
+  
+function convert(num) {
+    const hours = Math.floor(num / 60);
```
Comment (cecastosic) [inline javascript/javascript1/week3/whenWillWeBeThere.js]: if you pass the whole object, how would you get the speed and the destination data?
Good reading https://www.educative.io/answers/what-is-object-destructuring-in-javascript

File: javascript/javascript1/week3/noteApp.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
+// NOTE TAKING APP
+
+// Save a note
+const notes = [];
+
+function saveNote(content, id) {
+    notes.push({content, id});
```
Comment (cecastosic) [inline javascript/javascript1/week3/noteApp.js]: You should check here, before pushing if 
```
content parameter is a string
id is a number
```

File: javascript/javascript1/week3/noteApp.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+saveNote('Pick up groceries', 1);
+saveNote('Do laundry', 2);
+console.log(notes);
+
+
+// Get a note
+function getNote(id) {
+    if  (typeof id === '' || typeof id === 'null' || typeof id !== 'number') {
```
Comment (cecastosic) [inline javascript/javascript1/week3/noteApp.js]:
```suggestion
    if  (typeof id !== 'number') {
```

File: javascript/javascript1/week3/noteApp.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        alert(`Error: There is no note corresponding to this number. Please enter a valid number.`);
+    } else {
+        for (i = 0; i < notes.length; i++) {
+            if (notes[i].id === id) {
```
Comment (cecastosic) [inline javascript/javascript1/week3/noteApp.js:23]: Check find https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: javascript/javascript1/week3/noteApp.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+logOutNotesFormatted(notes);
+
+
+// Unique feature
+// Replacing the note's content
+function replaceNote(content, id) {
+    for (i = 0; i < notes.length; i++) {
+        if (notes[i].id === id) {
+            notes[i].content = content;
+        } else if (typeof id === '' || typeof id === 'null' || typeof id !== 'number' || id > notes.length || id < 1) {
```
Comment (cecastosic) [inline javascript/javascript1/week3/noteApp.js]: if the id is a type number, it can't be `null` or `""`

File: javascript/javascript1/week3/seriesDuration.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+const myFavoriteSeries = {title: 'Stranger Things', days: 1, hours: 4, minutes: 20};
+seriesDurations.push(myFavoriteSeries);
+  
+console.log(seriesDurations);
+let percentageOfLife = 0;
+
+const calcSeriesTime = function() {
+    const lifeSpanInMin = 80 * 365 * 60 * 24;
+    const dayToMinutes = 24 * 60;
+    const hoursToMinutes = 60;
```
Comment (cecastosic) [inline javascript/javascript1/week3/seriesDuration.js]: Nice! I would also add the `age` variable and give value of `80`

**Review by cecastosic — COMMENTED** [review]
Great job @RodrigoGuinsbergPinto 💪 I added some ideas for refactoring 🌸

File: javascript/javascript1/week2/js-week2.js (hunk 1 above)
Comment (RodrigoGuinsbergPinto) [reply javascript/javascript1/week2/js-week2.js]: Battling with git.
Learning from my mistakes.

File: javascript/javascript1/week3/js1-week3.html (hunk 2 above)
Comment (RodrigoGuinsbergPinto) [reply javascript/javascript1/week3/js1-week3.html]: Thanks.
I will use underscores (_) or dashes (-) next time.

**Review by Lokesh-sw — COMMENTED** [review]
@RodrigoGuinsbergPinto  looks good job with minor improvement opportunities.  @cecastosic have mentioned more than I possibly could. but in my opinion your getting logical ideas of using java script.

File: javascript/javascript1/week3/js1-week3.html (hunk 2 above)
Comment (cecastosic) [reply javascript/javascript1/week3/js1-week3.html]: Screenshot 2022-06-18 at 10 16 14
I am fine with files names (no need to use `-` or `_`). I want to say that you should write the file name as part of the src attribute. So instead of `<script src="item array removal.js"></script>`, based on the image above it should be `<script src="itemArrayRemoval.js"></script>`. Sorry if I wasn't clear.

File: javascript/javascript1/week3/noteApp.js (hunk 11 above)
Comment (RodrigoGuinsbergPinto) [reply javascript/javascript1/week3/noteApp.js]: Huumm, makes sense. : )

File: javascript/javascript1/week3/noteApp.js (hunk 8 above)
Comment (RodrigoGuinsbergPinto) [reply javascript/javascript1/week3/noteApp.js]: Ok.

File: javascript/javascript1/week3/noteApp.js (hunk 10 above)
Comment (RodrigoGuinsbergPinto) [reply javascript/javascript1/week3/noteApp.js:23]: I see what the find() method does, but I was not able to apply it to my exercise.
